python/samples/concepts/structured_output/json_structured_output.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import asyncio
+
+from semantic_kernel import Kernel
+from semantic_kernel.connectors.ai.function_choice_behavior import FunctionChoiceBehavior
+from semantic_kernel.connectors.ai.open_ai.services.azure_chat_completion import AzureChatCompletion
+from semantic_kernel.connectors.ai.open_ai.services.open_ai_chat_completion import OpenAIChatCompletion
+from semantic_kernel.contents import ChatHistory
+from semantic_kernel.contents.streaming_chat_message_content import StreamingChatMessageContent
+
+###################################################################
+# The following sample demonstrates how to create a chat          #
+# completion call that assists users in solving math problems.    #
+# The bot guides the user step-by-step through the solution       #
+# process using a structured output format based on either a      #
+# Pydantic model or a non-Pydantic model.                         #
+###################################################################
+
+
+###################################################################
+# NOTE: If using Azure OpenAI the the following is required:
+# - access to gpt-4o-2024-08-06
+# - the 2024-08-01-preview API version
+# - if using a token instead of an API KEY, you must have the
+#    `Cognitive Services OpenAI Contributor` role assigned to your
+#    Azure AD user.
```

Comment:
Question: just curious, is this true for all people, even external?

Reply:
I believe it is. Anyone that has setup token auth instead of key based auth on their AOAI instance will need the correct role on their user.